File: 2207-maximize-number-of-subsequences-in-a-string/2207-maximize-number-of-subsequences-in-a-string.java
class Solution {
    public long maximumSubsequenceCount(String text, String pattern) {
         long cnt1=0,cnt2=0,res=0;
        
        for(char t:text.toCharArray()){
            if(t==pattern.charAt(1)){
                res+=cnt1;cnt2++;
            }
            if(t==pattern.charAt(0))cnt1++;
        }
        return res + Math.max(cnt1,cnt2);
    }
}

File: 2207-maximize-number-of-subsequences-in-a-string/2207-maximize-number-of-subsequences-in-a-string.cpp
class Solution {
public:
    long long maximumSubsequenceCount(string text, string pattern) {
        
        long long cnt1=0,cnt2=0,res=0;
        
        for(char t:text){
            if(t==pattern[1]){
                res+=cnt1,cnt2++;
            }
            if(t==pattern[0])cnt1++;
        }
        return res + max(cnt1,cnt2);
    }
};